agent/dns.go

```diff
@@ -838,23 +859,33 @@ func (d *DNSServer) trimDomain(query string) string {
 	return strings.TrimSuffix(query, shorter)
 }
 
-// computeRCode Return the DNS Error code from Consul Error
-func (d *DNSServer) computeRCode(err error) int {
-	if err == nil {
+// rCodeFromError return the DNS Error code an error
+func rCodeFromError(err error) int {
+	switch {
+	case err == nil:
 		return dns.RcodeSuccess
-	}
-	if structs.IsErrNoDCPath(err) || structs.IsErrQueryNotFound(err) {
+	case errors.Is(err, errNoAnswer):
+		// TODO: why do we return success if the answer is empty?
```

Comment:
This is to avoid caching of error response for other query types. if we return an error a recursor will cache it and use it even for other query types

Reply:
Makes sense now that I've learned a bunch about DNS this week! I'll remove this TODO.